.github/workflows/ci.yml

```diff
@@ -63,6 +63,42 @@ jobs:
     - uses: bluefireteam/melos-action@v2
     - run: melos format
 
+  init-e2e:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: recursive
+    - uses: asdf-vm/actions/install@v3
+    - uses: bluefireteam/melos-action@v2
+    - name: Install dependencies
+      run: |
+        sudo apt update
+        sudo apt install -y clang cmake curl libgtk-3-dev ninja-build pkg-config unzip
+        sudo apt install -y gnome-initial-setup network-manager
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version-file: "./provd/go.mod"
+    - name: Build provd
+      run: go build ./cmd/provd
+      working-directory: provd
+    - name: Run provd
+      run: |
+        set -ux
+        set +e
+        sed 's|test.sock|provd/provd.sock|' provd/example.config.yaml  > provd/provd.yaml
```

Comment:
Maybe it is worth updating  `example.comfig.yaml ` / creating a new one specifically for the e2e tests? Will make this more robust in the event someone accidentally changes the example config and not update the e2e test?

Reply:
Updating the example config (and maybe renaming it to 'default') sounds like a good idea.

Reply:
Sounds perfect :+1:

Reply:
Hm, since provd expects a `provd.yaml` renaming 'example' to 'default' is probably misleading, so I'd just update the path